packages/core/src/inputRules/wrappingInputRule.ts

```diff
@@ -45,6 +51,24 @@ export function wrappingInputRule(config: {
 
       tr.wrap(blockRange, wrapping)
 
+      if (config.keepMarks && config.editor) {
+        const { selection, storedMarks } = state
+        const { splittableMarks } = config.editor.extensionManager
+        const marks = storedMarks || (selection.$to.parentOffset && selection.$from.marks())
+
+        if (marks) {
+          const filteredMarks = marks.filter(mark => splittableMarks.includes(mark.type.name))
+
+          tr.ensureMarks(filteredMarks)
+        }
+      }
+      if (config.keepAttributes) {
+        /** If the nodeType is `bulletList` or `orderedList` set the `nodeType` as `listItem` */
+        const nodeType = config.type.name === 'bulletList' || config.type.name === 'orderedList' ? 'listItem' : 'taskList'
```

Comment:
@bdbch I'm just worried about this piece of code, please let me know if there's any better way of doing this. The idea here is 
- When there's a paragraph with say colored text
- When we switch over to 2nd line, we apply the color to the whole `<li>` only then it is color is also applied for the `list-style` indicator.